### PR TITLE
feat(condo): DOMA-6556 call discoverServiceConsumers when creating Re…

### PR DESCRIPTION
**Pull request by abshnko**
…sident and BillingAccount

### CODE REVIEW

File: apps/condo/domains/billing/schema/BillingAccount.js [hunk 1]
```diff
@@ -101,6 +103,16 @@ const BillingAccount = new GQLListSchema('BillingAccount', {
                 return addValidationError(`${UNEQUAL_CONTEXT_ERROR}:property:context] Context is not equal to property.context`)
             }
         },
+        afterChange: async ({ context, updatedItem, operation }) => {
+            if (operation === 'create'){
```
Comment (SavelevMatthew) [inline apps/condo/domains/billing/schema/BillingAccount.js:107]: Why there's no update? According to accesses we can freely change billing accounts.

Comment (SavelevMatthew) [reply apps/condo/domains/billing/schema/BillingAccount.js:107]: Should old service consumers be deleted if unitName changes for example, or should hook be called and so on?

Comment (SavelevMatthew) [reply apps/condo/domains/billing/schema/BillingAccount.js:107]: Also what should be done if BillingPorperty address changed?

**Comment by SavelevMatthew** [comment]
How should we migrate existing data?

File: apps/condo/domains/resident/gql.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ const RESIDENT_ORGANIZATION_FIELDS = 'id name country tin'
 const RESIDENT_PROPERTY_FIELDS = 'id name address'
 const ORGANIZATION_FEATURES_FIELDS = 'hasBillingData hasMeters'
 const PAYMENT_CATEGORIES_FIELDS = 'id categoryName billingName acquiringName'
-const RESIDENT_FIELDS = `{ user { id name locale } organization { id name tin country } residentOrganization { ${RESIDENT_ORGANIZATION_FIELDS} } property { id createdAt deletedAt } residentProperty { ${RESIDENT_PROPERTY_FIELDS} } address addressMeta { ${ADDRESS_META_SUBFIELDS_QUERY_LIST} } unitName unitType ${COMMON_FIELDS} organizationFeatures { ${ORGANIZATION_FEATURES_FIELDS} } paymentCategories { ${PAYMENT_CATEGORIES_FIELDS} } }`
+const RESIDENT_FIELDS = `{ user { id name locale phone } organization { id name tin country } residentOrganization { ${RESIDENT_ORGANIZATION_FIELDS} } property { id createdAt deletedAt } residentProperty { ${RESIDENT_PROPERTY_FIELDS} } address addressMeta { ${ADDRESS_META_SUBFIELDS_QUERY_LIST} } unitName unitType ${COMMON_FIELDS} organizationFeatures { ${ORGANIZATION_FEATURES_FIELDS} } paymentCategories { ${PAYMENT_CATEGORIES_FIELDS} } }`
```
Comment (dkoviazin) [inline apps/condo/domains/resident/gql.js:20]: phone is a private data - better to use getById where you need phone and do not add this field to common queries

File: apps/condo/domains/billing/schema/BillingAccount.js (hunk 1 above)
Comment (dkoviazin) [reply apps/condo/domains/billing/schema/BillingAccount.js:107]: There will be no cases when address for billingProperty will change
And discoverService consumer will not watch for serviceConsumers after it creates them

Comment (dkoviazin) [reply apps/condo/domains/billing/schema/BillingAccount.js:107]: we want discoverServiceConsumer to work only on new residents and new billing accounts
it's not an universal tool for managing service consumers for resident 
it's only a helper to get receipts for the first time

Comment (SavelevMatthew) [reply apps/condo/domains/billing/schema/BillingAccount.js:107]: Should we block update access on billing entities then?

**Review by SavelevMatthew — COMMENTED** [review]
You should add more `real life` cases like:
```
1. Organization uploaded receipts
2. Resident registered
3. Resident can see receipts
```

```
1. Redisent registered
2. Receipts uploaded for a first time
3. Resident can see receipts
```

**Comment by AleX83Xpert** [comment]
> How should we migrate existing data?

There will be a script for starting this mutation with different arguments.

**Pull request closed without merging**